overlord/snapstate/snapstate.go

```diff
@@ -78,6 +82,17 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 		return nil, err
 	}
 
+	// ensure core gets installed. if it is already installed return
+	// an empty task set
+	ts := state.NewTaskSet()
+	if flags&doNotAutoInstallMissingBases == 0 {
+		var err error
+		ts, err = ensureCore(st, snapsup.Name(), snapsup.UserID)
```

Comment:
I think this is too simple,  because during an InstallMany or UpdateMany we will get here many times but the change doesn't exist yet, so will create many tasks to install core unless I'm confused